app/email.py

```diff
@@ -0,0 +1,43 @@
+from datetime import datetime
+from models.request_models.email import Email as EmailRequest
+from app.notify import send_email as send_email_to_notify
+from config import Config
+
+
+class Email(EmailRequest):
+    """
+    This is the Email model, which defines what an email object on the message queue looks like.
+    It is initalised based upon an EmailReqest, which defines the structure of an incoming payload.
+    """
+
+    origin_time: datetime = None
+    retry_count: int = 0
+
+    def __init__(self, email_request: EmailRequest):
+        if not email_request.personalisation:
+            email_request.personalisation = {}
+
+        super().__init__(
+            email_address=email_request.email_address,
+            template_id=email_request.template_id,
+            personalisation=email_request.personalisation,
+        )
+
+    def send_email(self):
+        send_email_to_notify(self.email_address, self.template_id, self.personalisation)
+
+    def get_retry_time_seconds(self) -> int:
```

Comment:
Should this moved to the task as it's about how to retry the task rather than the email request

Reply:
Hi Said, I agree, I have moved this.

Many thanks! 😄